File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/common/eventstore/AggregateAndEventIdsInLastEvent.java
package cloud.ambar.common.eventstore;

import cloud.ambar.common.aggregate.Aggregate;
import lombok.Getter;

@Getter
public class AggregateAndEventIdsInLastEvent {
    private final Aggregate aggregate;
    private final String eventIdOfLastEvent;
    private final String correlationIdOfLastEvent;

    public AggregateAndEventIdsInLastEvent(Aggregate aggregate, String eventIdOfLastEvent, String correlationIdOfLastEvent) {
        this.aggregate = aggregate;
        this.eventIdOfLastEvent = eventIdOfLastEvent;
        this.correlationIdOfLastEvent = correlationIdOfLastEvent;
    }
}

File: application/backend-credit-card-enrollment/backend-c#/Common/EventStore/AggregateAndEventIdsInLastEvent.cs
namespace CreditCardEnrollment.Common.EventStore;

public class AggregateAndEventIdsInLastEvent
{
    public required Aggregate.Aggregate Aggregate { get; init; }
    public required string EventIdOfLastEvent { get; init; }
    public required string CorrelationIdOfLastEvent { get; init; }
}